.github/workflows/CI-orchestrator.yml

```diff
@@ -0,0 +1,170 @@
+name: Orchestrator
+
+on:
+  pull_request:
+    branches: 
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  build-test-job:
+    uses: ./.github/workflows/CI-build-test.yml
+  test-coverage-job:
+    uses: ./.github/workflows/CI-coverage.yml
+  lint-format-job:
+    uses: ./.github/workflows/CI-lint-format.yml
+  e2e-test-job:
+    uses: ./.github/workflows/CI-e2e-test.yml
+
+  notify-slack:
+    runs-on: ubuntu-latest
+    needs: [build-test-job,test-coverage-job,lint-format-job, e2e-test-job]
+    if: ${{ always() }}
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v4
+      - name: Report Job Statuses
+        run: |
+          echo "Build and Test Job Status: ${{ needs.build-test-job.result }}"
+          echo "Test Coverage Job Status: ${{ needs.test-coverage-job.result }}"
+          echo "Lint and Format Job Status: ${{ needs.lint-format-job.result }}"
+          echo "E2E Test Job Status: ${{ needs.e2e-test-job.result }}"
+      - name: Download All Artifacts
+        uses: actions/download-artifact@v4
+      - name: Display structure of downloaded files
+        run: ls -R
+      - name: Summarize Test Data
+        run: ./.github/scripts/summarize_tests.sh
+      - name: Display unit test output
+        run: |
+          if [ -f "./unit-test-output/unit_tests_output.txt" ]; then
+            cat "./unit-test-output/unit_tests_output.txt"
+          else
+            echo "Unit test output not found."
+          fi
+      - name: Display integration test output
+        run: |
+          if [ -f "./integration-test-output/integration_tests_output.txt" ]; then
+            cat "./integration-test-output/integration_tests_output.txt"
+          else
+            echo "Integration test output not found."
+          fi
+      - name: Display test coverage output
+        run: |
+          if [ -f "./coverage-output/coverage_report.json" ]; then
+            cat "./coverage-output/coverage_report.json"
```

Comment:
I don't know if this is possible, but could piping this `cat` with `jq` eventually improve readability of the report?

Reply:
Thanks Alessandro! It's a good question, maybe it is an improvement for the future if required